src/mono/mono/sgen/sgen-marksweep.c

```diff
@@ -49,7 +49,13 @@ static int ms_block_size;
  * Don't allocate single blocks, but alloc a contingent of this many
  * blocks in one swoop.  This must be a power of two.
  */
+#ifndef TARGET_WASM
 #define MS_BLOCK_ALLOC_NUM	32
+#else
+// WASM doesn't support partial munmap, so we can't free individual blocks
+// we use posix_memalign and free the whole block at once
+#define MS_BLOCK_ALLOC_NUM	1
```

Comment:
we should investigate whether this will meaningfully regress performance; mwpm caused a benchmark regression that was fairly sizable

Reply:
Do you remember what specifically was impacted ?

Reply:
it was a stack walking benchmark in the performance suite i believe, and it was caused by the lock free allocator not properly utilizing all the space it allocated

Reply:
I made `LOCK_FREE_ALLOC_SB_MAX_SIZE` to `16384 * 32` to compensate for `MS_BLOCK_ALLOC_NUM 1`